modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Alternative.kt

```diff
@@ -18,15 +20,15 @@ interface Alternative<F> : Applicative<F>, MonoidK<F> {
    * @receiver computation to repeat.
    * @returns the collection of results with at least 1 repetition.
    */
-  fun <A> Kind<F, A>.some(): Kind<F, SequenceK<A>> = lazyAp { many().map { { a: A -> (sequenceOf(a) + it).k() } } }
+  fun <A> Kind<F, A>.some(): Kind<F, NonEmptyList<A>> = map { a: A -> { xs: ListK<A> -> Nel(a, xs) } }.lazyAp { many() }
```

Comment:
These type signatures fit better, I ran into those because lazyAp changed and noticed that there is no point using `Sequence` (also using + there can lead to stackoverflows). This is potentially infinite, but you can't access the return value anyway in those cases because the effect of F is getting applied forever before you even get there.